### PR TITLE
feat: add file modification policy + UI

### PR DESCRIPTION
**Pull request by carlinmack**
:heart: Thank you for your contribution!

### Description

> Please describe briefly your pull request.

Following the work on record deletion, we're now extending a similar flow to file modification. For record deletion we supported both immediate and request deletion, but for file modification we're only planning to implement immediate file modification and the request flow will be handled by instance admins.

image

**Expanded**

image

**Modal**

image

### Checklist

Ticks in all boxes and 🟢 on all GitHub actions status checks are required to merge:

- [ ] I'm aware of the [code of conduct](https://inveniordm.docs.cern.ch/contribute/code-of-conduct/).
- [ ] I've created [logical separate commits](https://inveniordm.docs.cern.ch/community/code/best-practices/commits/#commits) and followed the [commit message format](https://inveniordm.docs.cern.ch/community/code/best-practices/commits/#commit-message).
- [ ] I've added relevant test cases.
- [ ] I've added relevant documentation.
- [ ] I've marked [translation strings](https://inveniordm.docs.cern.ch/community/translations/i18n/).
- [ ] I've identified the [copyright holder(s)](https://inveniordm.docs.cern.ch/community/copyright-policy/) and updated copyright headers for touched files (>15 lines contributions).
- [ ] I've NOT included third-party code (copy/pasted source code or new dependencies).
    * If you have added [third-party code (copy/pasted or new dependencies)](https://inveniordm.docs.cern.ch/community/code/best-practices/commits/#third-party-codedependencies), please reach out to an [architect on Discord](https://discord.gg/8qatqBC).

**Frontend**

- [ ] I've followed the [CSS/JS](https://inveniordm.docs.cern.ch/community/code/best-practices/css-js/) and [React](https://inveniordm.docs.cern.ch/community/code/best-practices/react/) guidelines.
- [ ] I've followed the [web accessibility](https://inveniordm.docs.cern.ch/community/code/best-practices/accessibility/) guidelines.
- [ ] I've followed the [user interface](https://inveniordm.docs.cern.ch/community/code/best-practices/ui/) guidelines.

**Reminder**

By using GitHub, you have already agreed to the [GitHub’s Terms of Service](https://help.github.com/articles/github-terms-of-service/#6-contributions-under-repository-license) including that:

1. You license your contribution under the same terms as the current repository’s license.
2. You agree that you have the right to license your contribution under the current repository’s license.

### CODE REVIEW

File: invenio_rdm_records/assets/semantic-ui/js/invenio_rdm_records/src/deposit/fields/FileUploader/FileModification/ModificationModal.js [hunk 1]
```diff
@@ -0,0 +1,246 @@
...
+    }
+  };
+
+  render() {
+    const { open, fileModification } = this.props;
+    const { loading, error, messages, checklistState, checkboxState } = this.state;
+
+    const formDisabled =
+      checkboxState.some((v) => v === false) ||
+      checklistState.some((x) => x === true || x === undefined);
```
Comment (kpsherva) [inline invenio_rdm_records/assets/semantic-ui/js/invenio_rdm_records/src/deposit/fields/FileUploader/FileModification/ModificationModal.js]: it is a little bit ambiguous what is the logic to disable the form. could you add a comment ?

also I think formik should be able to handle this in its state if the checkbox fields are required, why do you manually do the form validation?

File: invenio_rdm_records/services/deletion_policy.py
[the remark this answers is not in the document]
Comment (ntarocco) [reply invenio_rdm_records/services/deletion_policy.py]: `request_policy.py`? or `requests_policies.py`?

**Review by ntarocco — COMMENTED** [review]
I feel a bit lost with all that React code for such a simple form, but I assume that it is needed.
The backend LGTM, however more tests would be appreciated, in particular to check corner cases related to permissions.

For the UI, it looks nice, I am not convinced by the light blue section "Edit files". I am wondering how visible this feature should be.
E.g. GitHub uses simple menus for some features. Is it maybe something to take into account?
Screenshot 2025-10-28 at 17 39 44

File: invenio_rdm_records/assets/semantic-ui/js/invenio_rdm_records/src/deposit/fields/FileUploader/FileModification/RadioGroup.js
[the remark this answers is not in the document]
Comment (kpsherva) [reply invenio_rdm_records/assets/semantic-ui/js/invenio_rdm_records/src/deposit/fields/FileUploader/FileModification/RadioGroup.js]: alright, in this case it would need to be renamed, because it does something really specific - displays Yes/No options so it is not generic `RadioGroup` component

**Comment by kpsherva** [comment]
Regarding the UI, I have a couple of comments:

Screenshot 2025-11-07 at 14 12 35

I think we shouldn't use message component to pass possible interaction options inside, since everywhere in the interface we are using the message component only to communicate simple information, without any interactions possible.

I would instead maybe use a modal with a choice or a button? we could discuss possible options

Screenshot 2025-11-07 at 14 12 38

The user will not know what "locking files" means - it is a term we use internally - not present anywhere else in the interface, so we should
```diff
- To correct minor errors, you can unlock the files
+ To correct minor errors, you can edit published files
```
To attach the button visually with the description, it would be better to go with this structure:

Screenshot 2025-11-07 at 14 29 53

The least preferred action option (edit published files) should be in more neutral color, grey preferably, so that we do not promote visually file edition.

File: invenio_rdm_records/assets/semantic-ui/js/invenio_rdm_records/src/deposit/fields/FileUploader/EditFilesAccordion.js [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+  };
+
+  render() {
+    const { record, permissions, fileModification } = this.props;
+    const { activeIndex } = this.state;
+    return (
+      <Message info>
+        <Accordion className="m-0">
+          <Accordion.Title
+            className="ui"
```
[the remark this answers is not in the document]
Comment (carlinmack) [reply invenio_rdm_records/assets/semantic-ui/js/invenio_rdm_records/src/deposit/fields/FileUploader/EditFilesAccordion.js]: I agree, but the `ui` class still needs to be added so that `color: #fff` etc aren't applied. Perhaps `.ui.accordion.invenio-accordion-field > .title:not(.ui)`?

(for context for others, this is the PR being discussed here https://github.com/inveniosoftware/invenio-app-rdm/pull/3200 )
